plugins/woocommerce/src/Blocks/BlockTypes/ProductCollection.php

```diff
@@ -80,8 +86,34 @@ protected function initialize() {
 		// Provide location context into block's context.
 		add_filter( 'render_block_context', array( $this, 'provide_location_context_for_inner_blocks' ), 11, 1 );
 
+		// Disable block render if the ProductTemplate block is empty.
+		add_filter(
+			'render_block_woocommerce/product-template',
+			function ( $html ) {
+				if ( null === $this->prevent_render ) {
+					$this->prevent_render = '' === $html;
+				}
+
+				return $html;
+			},
+			100,
+			1
+		);
+
+		// Enable block render if the ProductCollectionNoResults block is rendered.
+		add_filter(
+			'render_block_woocommerce/product-collection-no-results',
+			function ( $html ) {
+				$this->prevent_render = false;
+
+				return $html;
+			},
+			100,
+			1
+		);
```

Comment:
Not sure what you mean by "this doesn't work". 🤔 What should happen if there are no inner blocks in the "No results" block? Should we prevent the rendering of the entire PC block as well?

Reply:
> Should we prevent the rendering of the entire PC block as well?

IMO Yes. Because, In this case, HTML of no results block will be empty. Therefore, IMO It doesn't make sense to render PC block if:
- "Product Template" block is empty
- "No results" block is also empty

What do you think? 🤔

Reply:
Sounds valid to me. I wanted to make sure in case there's any other rendering aspect I missed. 👍

I guess it's also already covered by the proposed implementation [here](https://github.com/woocommerce/woocommerce/pull/50404/files#r1714984573), so no need to correct anything there.